### DISCUSSION TITLE
Is it okay if I use @remote-ui/vue for the host environment and @remote-ui/react for the remote environment?

### DISCUSSION THREAD

**Discussion by hamza-rd** [Shopify/remote-ui#241] [Q&A] [answered]
My web application relies on vue.js as the framework, necessitating the use of @remote-ui/vue for the host environment. However, I encountered an issue on the remote environment where @remote-ui/vue does not offer intellisense as @remote-ui/react does due to its direct support for JSX. Additionally, the latest tools do not seamlessly support vue.js with JSX bundling, in contrast to react JSX where esbuild works effortlessly out of the box.

**Comment by lemonmade** [comment, marked as answer]
Hi @hamza-rd, sorry for missing your message here, and the issues you opened. As far as I know, no one actively used the `@remote-ui/vue` package, which you could probably have guessed based on its poor quality and us never actively improving it. I wrote this package as a proof of concept, but never properly maintained it, and I'm sorry for that.

To answer part of your original question: yes, it’s no problem to mix-and-match JavaScript libraries on the remote and host side of a remote-ui project.

You may not need any of this anymore, but we've just completely rewritten `remote-ui` in a way that may be interesting to you. It’s now [entirely based on the DOM](https://github.com/Shopify/remote-dom/discussions/267), and since Vue has good support for custom elements (the building blocks of the new library), it works without any additional configuration on the ”remote” side. Because it’s based on the DOM, the remote side can actually use a much wider variety of JavaScript libraries than were possible before.

As part of this refactor, we’ve removed the `@remote-ui/vue` package entirely. This means that, for your “host” application written in Vue, there is no longer any utility for mapping the remote elements to Vue components. However, you can use the [`RemoteReceiver`](https://github.com/Shopify/remote-dom/tree/main/packages/core#remotereceiver), [`DOMRemoteReceiver`](https://github.com/Shopify/remote-dom/tree/main/packages/core#domremotereceiver), or [`SignalRemoteReceiver`](https://github.com/Shopify/remote-dom/tree/main/packages/signals#signalremotereceiver) to store the state of the remote tree, and map that state to Vue components yourself.
